### PR TITLE
Add `label_smoothing` param in `nn.BCELoss` and `nn.BCEWithLogitsLoss`

### LINKED ISSUE #91545: Adding label smoothing option to `nn.BCELoss`  and `nn.BCEWithLogitsLoss`?
### 🚀 The feature, motivation and pitch

Hello, I was looking at BCELoss and CrossEntropyLoss and found that while the latter has the option to add label_smoothing, the former does not. 

### Alternatives

Seeing as how the reasoning used for performing label smoothing (i.e. the logit values having to go to $\pm\infty$ to approach 1 or 0) should also apply to sigmoid functions (which are used with BCELoss), it stands to reason that label smoothing can and should be implemented with BCELoss? (i.e. for ex :  if the label value is 1, with label smoothing of 0.2, the value should be 0.8?)

### Additional context

_No response_

cc @albanD @mruberry @jbschlosser @walterddr @kshitij12345 @saketh-are

### PR DESCRIPTION
**Pull request by zeshengzong**
Fixes #91545

## Changes

- Add `label_smoothing` param and docs
- Add test case for `label_smoothing`
- Remove duplicate description in `nn.BCELoss` and `nn.BCEWithLogitsLoss`

##  Test Result

```bash
pytest -s test/test_nn.py -k test_bce
```

image

image

image

### CODE REVIEW

**Review by cyyever — DISMISSED** [review]

**Comment by zeshengzong** [comment]
Hello @jbschlosser @mikaylagawarecki, please help review this PR, thanks!

**Review by jbschlosser — COMMENTED** [review]
Python side impl looks good to me, thanks for the contribution :)

I will leave it to torch.nn maintainer @mikaylagawarecki for the final stamp though!

**Comment by mikaylagawarecki** [comment]
Hi @zeshengzong I'll leave a proper review on this before the end of the week. Thank you for your patience and your multiple contributions!

File: torch/nn/functional.py [hunk 1]
```diff
@@ -3490,31 +3490,11 @@ def binary_cross_entropy(
     size_average: Optional[bool] = None,
     reduce: Optional[bool] = None,
     reduction: str = "mean",
+    label_smoothing: float = 0.0,
 ) -> Tensor:
     r"""Compute Binary Cross Entropy between the target and input probabilities.
 
     See :class:`~torch.nn.BCELoss` for details.
-
```
Comment (mikaylagawarecki) [inline torch/nn/functional.py:3526]: Hm, why delete the args here and below? The convention in this file seems to be to document the args even though there's the see :class: ... on all the ops

File: test/test_nn.py [hunk 2]
```diff
@@ -4697,6 +4697,30 @@ def test_bce_loss_size_mismatch(self):
...
+        label_smoothings = [0.05, 0.15]
+
+        for loss_fn, label_smoothing in product([nn.BCELoss, nn.BCEWithLogitsLoss], label_smoothings):
+            loss = loss_fn(label_smoothing=label_smoothing)
+            output_with_smoothing = loss(inputs, target)
+            target_with_smoothing = target * (1 - label_smoothing) + (1 - target) * label_smoothing
+            loss = loss_fn()
+            output_with_manual_smoothing = loss(inputs, target_with_smoothing)
+            self.assertEqual(output_with_smoothing, output_with_manual_smoothing)
+
```
Comment (mikaylagawarecki) [inline test/test_nn.py]: would be nice to update the ModuleInputs instead :) (test/test_nn.py is somewhat legacy)

https://github.com/pytorch/pytorch/blob/2247aa6d1d43e256255f5c74a781c3190a4387b6/torch/testing/_internal/common_modules.py#L1455-L1542

**Review by mikaylagawarecki — COMMENTED** [review]
just nit comments, thanks!

File: test/test_nn.py (hunk 2 above)
Comment (zeshengzong) [reply test/test_nn.py]: Changed, thanks!

File: torch/nn/functional.py (hunk 1 above)
Comment (zeshengzong) [reply torch/nn/functional.py:3526]: Hi, previously I saw a comment said documents of [torch.nn.functional](https://docs.pytorch.org/docs/stable/nn.functional.html) methods intentionally left empty to avoid duplication with class document, like [torch.nn.functional.adaptive_avg_pool1d](https://docs.pytorch.org/docs/stable/generated/torch.nn.functional.adaptive_avg_pool1d.html#torch.nn.functional.adaptive_avg_pool1d), [torch.nn.functional.adaptive_avg_pool2d](https://docs.pytorch.org/docs/stable/generated/torch.nn.functional.adaptive_avg_pool2d.html#torch.nn.functional.adaptive_avg_pool2d), [torch.nn.functional.adaptive_avg_pool3d](https://docs.pytorch.org/docs/stable/generated/torch.nn.functional.adaptive_avg_pool3d.html#torch.nn.functional.adaptive_avg_pool3d), but there are some methods do have param documents like this one. 

I think it would be better to consistent content of `torch.nn.functional` documents (either all have param doc, or all left empty and guide user to class doc), avoid people think it was a mistake for those don't have param docs. But I'm not sure which is the right way to fix them. WDYT? Thanks!

Comment (mikaylagawarecki) [reply torch/nn/functional.py:3526]: oh, hmm I'm not sure what you mean as it seems the Args are still documented for `adaptive_avg_pool1d` 

https://github.com/pytorch/pytorch/blob/d5966249be93cb7b1f70bb8be24fa760948239d5/torch/nn/functional.py#L1357-L1358

I think it would be good not to delete these. If there is an inconsistent convention we can resolve that in a separate PR

**Review by mikaylagawarecki — DISMISSED** [review]
Looks good, please run lint again and remove the Args deletion for now :)

**Comment by zeshengzong** [comment]
> fyi, the internal failure looks expected and seems an internal test just needs updating, will reland after confirming

Do I need create another PR since this has been reverted?

**Comment by waylonflinn** [comment]
Pinging here, to hopefully get some attention on this PR. It has a feature that would be useful to me and has already been reviewed, approved and merged, but was reverted at some point because of an internal test.

Any way to get this merged again?
